lib/cudadrv/synchronization.jl

```diff
@@ -133,15 +134,25 @@ function synchronization_worker(data)
 end
 
 @noinline function create_synchronization_worker(i)
-    sync_channels[i] = BidirectionalChannel{SyncObject,CUresult}()
-    # should be safe to assign before threads are running;
-    #  any user will just submit work that makes it block
+    lock(sync_channel_lock) do
+        # test and test-and-set
+        if isassigned(sync_channels, i)
+            return
+        end
+
+        # should be safe to assign before threads are running;
+        # any user will just submit work that makes it block
+        sync_channels[i] = BidirectionalChannel{SyncObject,CUresult}()
 
-    # we don't know what the size of uv_thread_t is, so reserve enough space
-    tid = Ref{NTuple{32, UInt8}}(ntuple(i -> 0, 32))
+        # we don't know what the size of uv_thread_t is, so reserve enough space
```

Comment:
FWIW, uv_thead_t is a Csize_t AFAIK, and requires alignment as such, which this does not have